RiskAssessmentApi.yaml

```
@@ -276,24 +590,115 @@ components:
+              format: int32
+              description: The number of waypoints receiving a HIGH risk
+            mediumRiskWaypointCount:
+              type: integer
+              format: int32
+              description: The number of waypoints receiving a MEDIUM risk
+            lowRiskWaypointCount:
+              type: integer
+              format: int32
+              description: The number of waypoints receiving a LOW risk
+            obstacleProximityRiskWaypointPercentage:
+              type: number
+              format: double
+              minimum: 0.0
+              maximum: 100.0
+              description: The percentage of waypoints receiving a HIGH or MEDIUM obstacle proximity risk
+            groundImpactRiskWaypointPercentage:
+              type: number
+              format: double
+              minimum: 0.0
+              maximum: 100.0
+              description: The percentage of waypoints receiving a HIGH or MEDIUM ground impact risk
+            navigationReliabilityRiskWaypointPercentage:
+              type: number
+              format: double
+              minimum: 0.0
+              maximum: 100.0
+              description: The percentage of waypoints receiving a HIGH or MEDIUM navigation reliability risk
+          additionalProperties: true
+        riskResults:
           type: object
+          description: The risk assessment results used to determine the risk levels
+          properties:
+            overallRisk:
+              $ref: '#/components/schemas/RiskResults'
+              description: The mapping of waypoint number to overall risk level
+            obstacleProximityRisk:
+              $ref: '#/components/schemas/RiskResults'
+              description: The mapping of waypoint number to obstacle proximity risk level
+            groundImpactRisk:
+              $ref: '#/components/schemas/RiskResults'
+              description: The mapping of waypoint number to ground impact risk level
+            navigationReliabilityRisk:
+              $ref: '#/components/schemas/RiskResults'
+              description: The mapping of waypoint number to navigation reliability risk level
           additionalProperties: true
-          description: The freeform risk assessment report
+        riskThresholds:
+          $ref: '#/components/schemas/RiskThresholds'
+          description: The thresholds used for this risk assessment.
+        flightOperation:
+          type: object
+          description: The flight operation unique identifier and waypoints for which the risk assessment report was generated
+          properties:
+            uuid:
+              type: string
+              format: uuid
+              description: The unique identifier of the flight operation
+            waypoints:
+              type: object
+              additionalProperties:
+                $ref: '#/components/schemas/Waypoint'
+              description: The mapping of waypoint number to waypoint definition 
```

Given that this is 'waypoints' and not 'waypoint', should this be an array instead of an object?
```suggestion
              type: array
              items:
                $ref: '#/components/schemas/Waypoint'
              minItems: 2
              description: An ordered list of waypoints making up the flight operation
```

---

The way this is defined is a map of waypoint number to waypoint object.
Example: {"1" : object, 2 : object, ... }

I wanted this to mirror the results for each risk type above which is also a map.

---

I see how that could be confusing. The report includes all waypoints from the flight operation not a subset. I'll have to check Angelo's prototype, but I think that they are numbered from 1 to n.